server/server.go

```diff
@@ -162,7 +165,9 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.LogSourceIPs, "server.log-source-ips-enabled", false, "Optionally log the source IPs.")
 	f.StringVar(&cfg.LogSourceIPsHeader, "server.log-source-ips-header", "", "Header field storing the source IPs. Only used if server.log-source-ips-enabled is true. If not set the default Forwarded, X-Real-IP and X-Forwarded-For headers are used")
 	f.StringVar(&cfg.LogSourceIPsRegex, "server.log-source-ips-regex", "", "Regex for matching the source IPs. Only used if server.log-source-ips-enabled is true. If not set the default Forwarded, X-Real-IP and X-Forwarded-For headers are used")
+	f.BoolVar(&cfg.LogRequestHeaders, "server.log-request-headers-enabled", false, "Optionally log request headers.")
```

Comment:
Right now I feel we do not want to add this flag for all `Server`s, and that it was deliberately omitted before.
Consumers of the library who do want to expose this config to their users can do it themselves and pass `Log`.

(I do not appear to have thought about this for `LogSourceIPsHeader`, etc., but here we are)